tests/darwin/importer/importer_test.py

```diff
@@ -0,0 +1,205 @@
+                mock_is_dir.return_value = True
+                mock_glob.return_value = [Path("example_dir/file1.txt"), Path("example_dir/file2.txt")]
+                result = _get_files_for_parsing([Path("example_dir")])
+                assert result == [Path("example_dir/file1.txt"), Path("example_dir/file2.txt")]
+
+    def test_get_files_for_parsing_single_file(self) -> None:
+        with patch.object(Path, "is_dir") as mock_is_dir:
+            with patch.object(Path, "glob") as mock_glob:
+                mock_is_dir.return_value = False
+
+                result = _get_files_for_parsing([Path("example_dir")])
+                assert result == [Path("example_dir")]
+                mock_glob.assert_not_called()
+
+
+class Test_find_and_parse(TestCaseImporter):
+    mock_console: Optional[MagicMock]
+
+    def setUp(self) -> None:
+        self.mock_console = MagicMock(spec=Console)
+        return super().setUp()
+
+    def tearDown(self) -> None:
+        self.mock_console = None
+        return super().tearDown()
+
+    @patch("darwin.importer.importer._get_multi_cpu_settings")
+    @patch("darwin.importer.importer._get_files_for_parsing")
+    @patch("darwin.importer.importer.WorkerPool")
+    def test_uses_mpire_if_use_multi_cpu_true(
+        self, mock_wp: MagicMock, mock_gffp: MagicMock, mock_gmcus: MagicMock
+    ) -> None:
+
+        from darwin.importer.importer import find_and_parse
+
+        mock_gmcus.return_value = (2, True)
+        mock_gffp.return_value = [Path("example_dir/file1.txt"), Path("example_dir/file2.txt")]
+
+        mock_importer = MagicMock()
+        mock_map = MagicMock()
+
+        class MockWorkerPool:
+            def __init__(self) -> None:
+                self.map = mock_map
+
+            def __enter__(self) -> "MockWorkerPool":
+                return self
+
+            def __exit__(self, *args) -> None:  # type: ignore
+                pass
+
+        mock_wp.return_value = MockWorkerPool()
+        mock_map.return_value = ["1", "2"]
+
+        result = find_and_parse(mock_importer, [Path("example_dir")], self.mock_console, True, 2)
+
+        mock_wp.assert_called_once()
+        mock_wp.assert_called_with(2)
+        mock_map.assert_called_once()
+
+        self.assertEqual(result, ["1", "2"])
+
+    @patch("darwin.importer.importer._get_files_for_parsing")
+    @patch("darwin.importer.importer.WorkerPool")
+    def test_runs_single_threaded_if_use_multi_cpu_false(self, mock_wp: MagicMock, mock_gffp: MagicMock) -> None:
+
+        from darwin.importer.importer import find_and_parse
+
+        mock_gffp.return_value = [Path("example_dir/file1.txt"), Path("example_dir/file2.txt")]
+
+        mock_importer = MagicMock()
+        mock_importer.side_effect = ["1", "2"]
+
+        result = find_and_parse(mock_importer, [Path("example_dir")], self.mock_console, False)
+
+        mock_wp.assert_not_called()
+        mock_importer.assert_called()
+
+        self.assertEqual(result, ["1", "2"])
```

Comment:
Just double checking that the result of `find_and_parse` is guaranteed to be in the same order always?

Reply:
With `map` yes, it will be - using `imap` or `starmap` it wouldn't necessarily be.